### ISSUE TITLE
cannot link: @java.io.OutputStreamWriter::append_java.lang.CharSequence_java.io.Writer

### ISSUE THREAD

**Issue by lihaoyi** [scala-native/scala-native#1280]
Not sure why this is the case, especially since this guy seems to exist:

- https://github.com/scala-native/scala-native/blob/master/javalib/src/main/scala/java/io/Writer.scala#L27

But this should be the last thing blocking me from porting uPickle over to scala-native

```text
$ mill hello.run
[35/64] hello.compile
[info] Compiling 1 Scala source to /Users/lihaoyi/test/out/hello/compile/dest/classes ...
[info] Done compiling.
[62/64] hello.nativeLink
Linking (1000 ms)
cannot link: @java.io.OutputStreamWriter::append_java.lang.CharSequence_java.io.Writer
1 targets failed
hello.nativeLink scala.scalanative.build.BuildException: unable to link
    scala.scalanative.build.Build$.build(Build.scala:60)
    mill.scalanativelib.bridge.ScalaNativeBridge.nativeLink(ScalaNativeBridge.scala:65)
    mill.scalanativelib.ScalaNativeModule.$anonfun$nativeLink$2(ScalaNativeModule.scala:173)
    mill.define.ApplyerGenerated.$anonfun$zipMap$1(ApplicativeGenerated.scala:5)
    mill.define.Task$MappedDest.evaluate(Task.scala:348)

$ cat hello/src/hello/Hello.scala
package hello
object Hello{
  def main(args: Array[String]): Unit = {
    new java.io.OutputStreamWriter(???).append("": CharSequence)
  }
}
```

The JVM implementation of OutputStreamWriter has this override:

```java
    @Override
    public Writer append(CharSequence csq) throws IOException {
        if (csq instanceof CharBuffer) {
            se.write((CharBuffer) csq);
        } else {
            se.write(String.valueOf(csq));
        }
        return this;
    }
```

Does scala-native need one in OutputStreamWriter as well?

**Comment by LeeTibbert** [comment]
@lihaoyi 
Thank you for reporting this problem.  Are you still experiencing it?

The reason that I ask is that I took a run at replicating the presenting problem
and OutputStreamWriter.append(CharSequence) seems to work on my
S-N 0.3.7 Ubuntu 18.04.1 X86 system just fine.  I will try on an up to date X96_64 system
running the latest 0.3.9-snapshot. 

I followed the commit history back to February 2018, and did not see anything obvious
that would fix or break the presenting problem.

As we all know, just one replication attempt fails does not mean that there isn't a problem.
Just that I have not been clever enough replicating.

My current hypothesis is a link resolution path issue.

If you are still experiencing a problem, could you do a run with the mill clang/linker option
for verbose ("-v") set on?  I'd like to see the list of object files & libraries being linked in.

I have not made the move to Mill yet, so I was using Sbt, with the standard ScalaNative 
build.sbt.  Perhaps I will try in a S-N project other than a clone of the S-N GitHub master.
Maybe there is some special link magic going on there.

I am keen to see uPickle and your other wonderful software ported to S-N, so I would
like to be able to help out here, if I can.

**Comment by LeeTibbert** [comment]
Just to establish a common point for bug hunting, the code that I am
using to try to replicate is below. I wrap it in just enough Scala so that
I can run it as a main program:

```
     val osw = new java.io.OutputStreamWriter(System.out)

    osw.append("Daisy, Daisy\n": CharSequence)

    osw.flush
```
System.out is of type `java.io.PrintStream`.  I just used it because it was
ready made.

I could try opening a file and creating another PrintStream if you think that
might be the worthwhile.

**Comment by densh** [comment]
Fixed in #1323
